tests/playbooks/tests_route_device.yml

```diff
@@ -0,0 +1,187 @@
+                type: ethernet
+                autoconnect: false
+                ip:
+                  address:
+                    - 198.51.100.6/24
+                    - 2001:db8::4/32
+                  route:
+                    - network: 198.51.12.128
+                      prefix: 26
+                      gateway: 198.51.100.1
+                      metric: 2
+        - name: Get the IPv4 routes from the route table main
+          command: ip -4 route
+          register: route_table_main_ipv4
+          changed_when: false
+
+        - name: Assert that the route table main contains the specified IPv4
+            routes
+          assert:
+            that:
+              # When running with nm provider, the route will be configured
+              # with `proto static`
+              # In RHEL-6.10 managed host, the attribute in the route is not
+              # equally spaced
+              - route_table_main_ipv4.stdout is search("198.51.10.64/26 via
+                198.51.100.6 dev ethtest0\s+(proto static )?metric 4")
+              - route_table_main_ipv4.stdout is search("198.51.12.128/26 via
+                198.51.100.1 dev ethtest1\s+(proto static )?metric 2")
+            msg: "the route table main does not contain the specified IPv4
+              route"
+
+        - name: Get the IPv6 routes from the route table main
+          command: ip -6 route
+          register: route_table_main_ipv6
+          changed_when: false
+
+        - name: Assert that the route table main contains the specified IPv6
+            routes
+          assert:
+            that:
+              - route_table_main_ipv6.stdout is search("2001:db6::4 via
+                2001:db8::1 dev ethtest0\s+(proto static )?metric 2")
+            msg: "the route table main does not contain the specified IPv6
+              route"
+        - name: Get the interface1 MAC address
+          command: cat /sys/class/net/{{ interface1 }}/address
+          register: interface1_mac
+          changed_when: false
+        - name: Configure the IP addresses and the route with only the MAC
+            address specified
+          import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              - name: "{{ interface1 }}"
+                mac: "{{ interface1_mac.stdout }}"
+                type: ethernet
+                autoconnect: false
+                ip:
+                  address:
+                    - 198.51.100.4/24
+                    - 2001:db8::6/32
+                  route:
+                    - network: 198.58.10.64
+                      prefix: 26
+                      gateway: 198.51.100.102
+                      metric: 4
+
+        - name: Assert that the warning about specifying the route without
+            the output device is logged for initscripts provider
+          assert:
+            that:
+              - __network_connections_result.stderr is search("<warn> .* The
```

Comment:
What is the wildcard matching here?

Reply:
The warning is:

```
stderr": "[003] <warn>  #0, state:None persistent_state:present, 'ethtest1': The connection 'ethtest1' does not specify an interface name. Therefore, the route '{'network': '198.58.10.64', 'prefix': 26, 'gateway': '198.51.100.102', 'metric': 4, 'table': None, 'family': <AddressFamily.AF_INET: 2>}' will be configured without the output device and the kernel will choose it automatically which might result in an unwanted device being used. To avoid this, specify `interface_name` in the connection appropriately.\n[004] <info>  #0, state:None persistent_state:present, 'ethtest1': update ifcfg-rh profile 'ethtest1'\n",
```

Reply:
You can mask problematic single characters with a dot, so for example:

`.003. <warn>  .0, state:None persistent_state:present, 'ethtest1':` - Assuming that the brackets or the hash sign caused the issues.

Reply:
Another "trick" is to define pattern strings as `vars` e.g.
```yaml
          assert:
            that:
              - __network_connections_result.stderr is search(__search_pattern)
          vars:
            __search_pattern: string that can have problematic meta characters
```

Reply:
The problematic characters are hash sign and colon.

Reply:
> The problematic characters are hash sign and colon. I updated. Thanks.

https://blogs.perl.org/users/tinita/2018/03/strings-in-yaml---to-quote-or-not-to-quote.html
"When not to use Plain Scalars"